fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/TrinoParserUtils.java

```diff
@@ -30,4 +42,59 @@ public static StatementBase toStatement(String query, long sqlMode) {
             throw new UnsupportedOperationException("Unsupported statement type: " + statement.getClass().getName());
         }
     }
+
+    /**
+     * Align with Trino's type conversion rules for Dates/Timestamps. For Trino, if the input type is Date,
+     * then the output type must also be Date:
+     * date '2012-08-08' + interval '2' day -> 2012-08-10;
+     * If the input type is Timestamp, then the output type should be Timestamp, as well:
+     * timestamp '2012-10-31 01:00' + interval '1' month -> 2012-11-30 01:00:00.000
+     * @param expr the input timestamp arithmetic expr which we need to determine the output type
+     * @return the wrapped or original expr, after applying appropriate output type conversions
+     */
+    public static Expr alignWithInputDatetimeType(TimestampArithmeticExpr expr) {
+        if (isDateType(expr.getChild(0))) {
+            return new CastExpr(Type.DATE, expr);
+        }
+        return expr;
+    }
+
+    /**
+     * Functions that return DATE as the result type
+     */
+    private static Set<String> DATE_RETURNING_FUNCTIONS = new HashSet<>();
+    static {
+        DATE_RETURNING_FUNCTIONS.add("date");
```

Comment:
You can add these functions to FunctionSet

Reply:
> You can add these functions to FunctionSet

OK, added, thanks